File: 231-PowerOfTwo.java
public class Solution {
    public static boolean isPowerOfTwo(int n) {
        if (n <= 0) return false;
        if (n == 1) return true;
        while (n % 2 ==0){
            if (n > 2)  n /= 2;
            else return true;
        }
        return (n % 2 == 0);
    }
}

File: cpp/231-PowerOfTwo.cpp
//Power of 2 means only one bit of n is '1', so use the trick n&(n-1)==0 to judge whether that is the case
class Solution {
public:
    bool isPowerOfTwo(int n) {
        return (n > 0) && (n & (n - 1)) == 0;
    }
};
